src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/ObjectIdLocalSymbol.cs

```diff
@@ -35,63 +36,46 @@ internal static BoundExpression RewriteLocal(CSharpCompilation compilation, EENa
-                    return new PlaceholderMethodSymbol(
-                        c,
-                        s,
-                        n,
-                        returnType,
-                        m => ImmutableArray.Create<ParameterSymbol>(new SynthesizedParameterSymbol(m, parameterType, ordinal: 0, refKind: RefKind.None)));
-                });
-            var getAddressMethod = container.GetOrAddSynthesizedMethod(
-                ExpressionCompilerConstants.GetVariableAddressMethodName,
-                (c, n, s) =>
-                {
-                    return new PlaceholderMethodSymbol(
-                        c,
-                        s,
-                        n,
-                        m => ImmutableArray.Create<TypeParameterSymbol>(new SimpleTypeParameterSymbol(m, 0, "<>T")),
-                        m => m.TypeParameters[0], // return type is <>T&
-                        m => ImmutableArray.Create<ParameterSymbol>(new SynthesizedParameterSymbol(m, parameterType, ordinal: 0, refKind: RefKind.None)),
-                        returnValueIsByRef: true);
-                });
             return new BoundPseudoVariable(
                 syntax,
                 local,
-                new ObjectIdExpressions(compilation, getValueMethod, getAddressMethod),
+                new ObjectIdExpressions(compilation),
                 local.Type);
         }
 
         private sealed class ObjectIdExpressions : PseudoVariableExpressions
         {
             private readonly CSharpCompilation _compilation;
-            private readonly MethodSymbol _getValueMethod;
-            private readonly MethodSymbol _getAddressMethod;
 
-            internal ObjectIdExpressions(CSharpCompilation compilation, MethodSymbol getValueMethod, MethodSymbol getAddressMethod)
+            internal ObjectIdExpressions(CSharpCompilation compilation)
             {
                 _compilation = compilation;
-                _getValueMethod = getValueMethod;
-                _getAddressMethod = getAddressMethod;
             }
 
             internal override BoundExpression GetValue(BoundPseudoVariable variable)
             {
+                var method = GetIntrinsicMethod(this._compilation, ExpressionCompilerConstants.GetVariableValueMethodName);
                 var local = variable.LocalSymbol;
-                var expr = InvokeGetMethod(_getValueMethod, variable.Syntax, local.Name);
+                var expr = InvokeGetMethod(method, variable.Syntax, local.Name);
                 return ConvertToLocalType(_compilation, expr, local.Type);
             }
 
             internal override BoundExpression GetAddress(BoundPseudoVariable variable)
             {
+                var method = GetIntrinsicMethod(this._compilation, ExpressionCompilerConstants.GetVariableAddressMethodName);
+                // Currently the MetadataDecoder does not support byref return types
```

Comment:
Could we avoid this problem by returning IntPtr?  I guess that's less desirable as a long-term solution.